### PR TITLE
OCPBUGS-24049:  Add option to enable/disable tailing to Pod log viewer mobile screen issues

### PR DESCRIPTION
**Pull request by cyril-ui-developer**
Follow on changes for [PR](https://github.com/openshift/console/pull/13298)

### Before:

Screenshot 2023-12-07 at 8 22 53 AM

Screenshot 2023-12-07 at 8 23 03 AM

### After:

Screenshot 2024-03-20 at 8 41 33 AM

Screenshot 2024-03-20 at 8 41 19 AM

Screenshot 2024-03-20 at 9 52 31 AM

### CODE REVIEW

**Comment by rhamilto** [comment]
I didn't review the code yet as I noticed a few concerns with the functionality:

There is an issue with links in the mobile dropdown where clicking the menu item instead of the link results in nothing happening.  Likely need to convert the links to actions rather than just putting them inside a menu item.

https://github.com/openshift/console/assets/895728/bd44a8df-baaf-4bcf-91ef-ea61a867fd23

Seems like the layout could be better here.  Since the right items are collapsing to a kebab, there is still plenty of room for the other items.  @sg00dwin, do you have time to help @cyril-ui-developer with optimizing the layout?
Screenshot 2023-12-07 at 3 15 38 PM

**Comment by cyril-ui-developer** [comment]
> I didn't review the code yet as I noticed a few concerns with the functionality:
> 
> There is an issue with links in the mobile dropdown where clicking the menu item instead of the link results in nothing happening. Likely need to convert the links to actions rather than just putting them inside a menu item.
> 
>  Screen.Recording.2023-12-07.at.3.14.26.PM.mov 
> Seems like the layout could be better here. Since the right items are collapsing to a kebab, there is still plenty of room for the other items. @sg00dwin, do you have time to help @cyril-ui-developer with optimizing the layout? Screenshot 2023-12-07 at 3 15 38 PM

@rhamilto I would have to add a click event to handle that for Download, Raw and Expand. If click event is added for full length of dropdown item, then  removing the underline would be idea for Download and Raw. WDYT?

**Comment by rhamilto** [comment]
> @rhamilto I would have to add a click event to handle that for Download, Raw and Expand. If click event is added for full length of dropdown item, then removing the underline would be idea for Download and Raw. WDYT?

My assumption is the links would get removed entirely, which would remove the blue and the underline from the text.  Is that what you mean?

**Comment by cyril-ui-developer** [comment]
> I didn't review the code yet as I noticed a few concerns with the functionality:
> 
> There is an issue with links in the mobile dropdown where clicking the menu item instead of the link results in nothing happening. Likely need to convert the links to actions rather than just putting them inside a menu item.
> 
>  Screen.Recording.2023-12-07.at.3.14.26.PM.mov 
> Seems like the layout could be better here. Since the right items are collapsing to a kebab, there is still plenty of room for the other items. @sg00dwin, do you have time to help @cyril-ui-developer with optimizing the layout? Screenshot 2023-12-07 at 3 15 38 PM

@rhamilto The UX design layout mobile screen in that order. Please double check the [story](https://issues.redhat.com/browse/CONSOLE-3732)

**Comment by cyril-ui-developer** [comment]
@rhamilto I have addressed the first comment, although this is not captured on the UX design but it looks good to me. See attached and PTAL. 
The second comment can be address but I am not sure if that is what we want since the UX design layout the mobile screen in that order. 
@jhadvig WDYT?

Screenshot 2023-12-08 at 7 20 56 AM
Screenshot 2023-12-08 at 7 21 16 AM

**Comment by cyril-ui-developer** [comment]
> > @rhamilto I would have to add a click event to handle that for Download, Raw and Expand. If click event is added for full length of dropdown item, then removing the underline would be idea for Download and Raw. WDYT?
> 
> My assumption is the links would get removed entirely, which would remove the blue and the underline from the text. Is that what you mean?

Hmm, `ConsoleExternalLogLink` is underline/blue color. Would try to fix it quickly.

**Comment by cyril-ui-developer** [comment]
> > > @rhamilto I would have to add a click event to handle that for Download, Raw and Expand. If click event is added for full length of dropdown item, then removing the underline would be idea for Download and Raw. WDYT?
> > 
> > 
> > My assumption is the links would get removed entirely, which would remove the blue and the underline from the text. Is that what you mean?
> 
> Hmm, `ConsoleExternalLogLink` is underline/blue color. Would try to fix it quickly.

@rhamilto @jhadvig But wait, if the "Raw, Download, Expand and ConsoleExternalLogLink" are links should they not be blue and underline  on hover in the Dropdown similar to the desktop screen? Maybe we have get the UX input on this? WDYT?

File: frontend/public/components/utils/resource-log.tsx [hunk 1]
```diff
@@ -286,118 +412,105 @@ export const LogControls: React.FC<LogControlsProps> = ({
...
+                    e.preventDefault();
+                    toggleWrapLines(!isWrapLines);
+                  }}
+                >
+                  {wrapLines}
+                </DropdownItem>
+                <DropdownItem to={currentLogURL} isExternalLink>
+                  {raw}
+                </DropdownItem>
+                <DropdownItem to={currentLogURL} isExternalLink>
```
Comment (rhamilto) [inline frontend/public/components/utils/resource-log.tsx:465]:
```suggestion
                <DropdownItem to={currentLogURL} download={`${resource.metadata.name}-${containerName}.log`>
```
Plus a forthcoming version of `@patternfly/react-core` that includes https://github.com/patternfly/patternfly-react/pull/9892

**Review by rhamilto — APPROVED** [review]
/lgtm

**Pull request merged**